### DISCUSSION TITLE
Version 2.1.4

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#53] [Announcements]
**Changes:**

- Force update interval of greater than 30 seconds as per new Olarm information.

Olarm had to implement a limit to their API endpoints as some users had set their update intervals too frequently causing a spike in usage and causing their API to not respond to a new request. Thus if you exceed your API key quota, your key will be blocked and you might experience other issues.

<hr /><em>This discussion was created from the release <a href='https://github.com/rainepretorius/olarm-ha-integration/releases/tag/'>Version 2.1.4</a>.</em>

**Comment by Chris-777** [comment]
Thank for all your work with this, very much appreciated! I have two devices on my account, an EVO192 and a MG5050. The API response looks normal but I'm typically getting these two errors with with 2.1.4 now after reinstalling the integration from scratch:

********* Error 1 *********
Logger: homeassistant.config_entries
Source: custom_components/olarm_sensors/__init__.py:283
Integration: Olarm Sensors ([documentation](https://github.com/rainepretorius/olarm-ha-integration/blob/main/README.md), [issues](https://github.com/rainepretorius/olarm-ha-integration/issues))
First occurred: 13:01:30 (1 occurrences)
Last logged: 13:01:30
Error setting up entry Olarm Sensors for olarm_sensors

Traceback (most recent call last):
  File "/config/custom_components/olarm_sensors/__init__.py", line 267, in update_listener
    config_entry.options[CONF_SCAN_INTERVAL] < 30
KeyError: 'scan_interval'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/olarm_sensors/__init__.py", line 46, in async_setup_entry
    await update_listener(hass, config_entry)
  File "/config/custom_components/olarm_sensors/__init__.py", line 283, in update_listener
    hass.config_entries.async_update_entry(config_entry, data=data, options=options)
UnboundLocalError: local variable 'data' referenced before assignment


********* Error 2 *********
Logger: homeassistant.config_entries
Source: custom_components/olarm_sensors/olarm_api.py:496
Integration: Olarm Sensors ([documentation](https://github.com/rainepretorius/olarm-ha-integration/blob/main/README.md), [issues](https://github.com/rainepretorius/olarm-ha-integration/issues))
First occurred: 1:11:45 PM (1 occurrences)
Last logged: 1:11:45 PM
Error setting up entry Olarm Sensors for olarm_sensors

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/olarm_sensors/__init__.py", line 70, in async_setup_entry
    devices = await setup_api.get_olarm_devices()
  File "/config/custom_components/olarm_sensors/olarm_api.py", line 496, in get_olarm_devices
    olarm_resp = await response.json()
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 1104, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8', url=URL('https://apiv4.olarm.co/api/v4/devices')

**Reply by maritzhome1** [reply]
Like Chris - many thanks for a very helpful and integration.

One comment from my end, each time I upgrade, after a restart the integration does not work. The only way to get it up and running again is to delete and reinstall and reconfigure. 

This happens for me with each update - so not unique to this version.

**Comment by rainepretorius** [comment]
Please note that this integration currently gives errors due to Olarm implementing a request rate limit. Thus the integration does not work currently until they develop their own Home Assistant solution or a WebSocket API. I will be able to give more information after I had a meeting with their CTO next week on the course of action forward.
